core/src/main/java/org/apache/iceberg/hadoop/HadoopInputFile.java

```diff
@@ -169,6 +174,24 @@ public FileStatus getStat() {
     return lazyStat();
   }
 
+  public Path getPath() {
+    return path;
+  }
+
+  public String[] getBlockLocations(long start, long end) {
+    List<String> hosts = Lists.newArrayList();
+    try {
+      for (BlockLocation location : fs.getFileBlockLocations(path, start, end)) {
+        Collections.addAll(hosts, location.getHosts());
+      }
+
+      return hosts.toArray(NO_LOCATION_PREFERENCE);
```

Comment:
Yeah, it is odd how this API works, but when you pass in an array with fewer elements than you need, it will allocate a new one for you that is the right size. So we pass in a 0-element array. If you just call `toArray`, Java won't get the type right. The streams API allows passing in an array constructor (`String[]::new`), but it seems awkward to create a stream when there's already a working, although awkward, API call for it.